### ISSUE TITLE
Interval not using Oracle specified Dialect

### ISSUE THREAD

**Issue by CaselIT** [sqlalchemy/sqlalchemy#10509]
### Discussed in https://github.com/sqlalchemy/sqlalchemy/discussions/10508

<div type='discussions-op-text'>

<sup>Originally posted by **indiVar0508** October 19, 2023</sup>
Hi
I was working on a issue and was getting incorrect create table query for oracle, please refer below 

```py
# If i do this and create Table
table = Table('my_interval_table', metadata,
    Column('my_interval_column', sa.Interval(day_precision=9, second_precision=9))
)
# Query generated for Oracle DB
CREATE TABLE my_interval_table (my_interval_column INTERVAL DAY TO SECOND)

# But if i directly get INTERVAL from oracle dialect 
table = Table('my_interval_table', metadata,
    Column('my_interval_column', sa.dialects.oracle.types.INTERVAL(day_precision=9, second_precision=9))
)
# GIves query
CREATE TABLE my_interval_table (my_interval_column INTERVAL DAY(9) TO SECOND(9))
```

but for postgresDB it seems to work fine, in [Interval Doc](https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.Interval) it talks about postgres but not Oracle DB, so wanted to confirm.

Thanks.</div>

**Comment by indiVar0508** [comment]
Hi @CaselIT ,
Raised a PR for this you can review once.

**Comment by sqla-tester** [comment]
**indivar** has proposed a fix for this issue in the **main** branch:

**fixes: #10509, use oracle.INTERVAL for generic interval** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4910

**Comment by sqla-tester** [comment]
**indivar** has proposed a fix for this issue in the **main** branch:

**use oracle.INTERVAL for generic interval** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4910
